src/oic/oauth2/util.py

```diff
@@ -57,7 +61,14 @@ def get_or_post(uri, method, req, content_type=DEFAULT_POST_CONTENT_TYPE,
     if method in ["GET", "DELETE"]:
         _qp = req.to_urlencoded()
         if _qp:
-            path = uri + '?' + _qp
+            comp = urlsplit(uri)
+            if comp.query:
+                req.update(parse_qs(comp.query))
+                _query = req.to_urlencoded()
+                path = urlunsplit((comp.scheme, comp.netloc, comp.path,
+                                  _query, comp.fragment))
+            else:
+                path = uri + '?' + _qp
```

Comment:
We can use the urlsplit/urlunsplit combo for this line as well. Since you are redoing this, it is a good idea to keep them the same.